### ISSUE TITLE
Lot of Pstates seems to not be used ? (7700k)

### ISSUE THREAD

**Issue by Pumpk1in** [Piker-Alpha/ssdtPRGen.sh#399]
Hi !

I have a 7700k CPU.
Using ssdtprgen, with that args : `/ssdtPRGen.sh -target 5 -turbo 4600 -bclk 100` because my CPU is overclocked.

The "problem" is that, even if ssdtprgen show me all the Pstates, MacOS seems to only use 4 of them. 1000, 1300, 4500 and 4600mhz.
As you can see here : 
![7700k](https://user-images.githubusercontent.com/28058282/32054658-7834019a-ba5f-11e7-8f56-94a5e3334d39.png)

Is that the correct behavior ?

Thanks !

**Comment by whatsnewsisyphus** [comment]
I seem to have this issue with native generator as well.
going between these three
`CPU P-States [ (8) 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 ] iGPU P-States [ 5 21 ]

CPU P-States [ 8 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 (42) 43 44 45 ] iGPU P-States [ 21 ]
CPU P-States [ 8 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 (45) ] iGPU P-States [ 5 21 ]

`

**Comment by Pumpk1in** [comment]
Yep, exactly the same. I see a lot of numbers, but only 4 states of them are used.
It's not really a problem on a desktop, we could be running it at 100% 24h a day but it's a strange behavior.

**Comment by whatsnewsisyphus** [comment]
I actually have slight hangups at times and have been wondering about what is going on with the speedstep
